src/Symfony/Component/Process/ExecutableFinder.php

```diff
@@ -69,7 +69,7 @@ public function find(string $name, string $default = null, array $extraDirs = []
         }
 
         $command = '\\' === \DIRECTORY_SEPARATOR ? 'where' : 'command -v';
-        if (\function_exists('exec') && ($executablePath = strtok(@exec($command.' '.escapeshellarg($name)), \PHP_EOL)) && @is_executable($executablePath)) {
+        if (\function_exists('exec') && !str_starts_with($name,'-') &&($executablePath = strtok(@exec($command.' '.escapeshellarg($name)), \PHP_EOL)) && @is_executable($executablePath)) {
```

Comment:
this looks wrong to me, if `$name` is not an executable it shouldn't be passed to `find()` in the first place

Reply:
In order to avoid users who do not understand the specific implementation of the following error, I think we should avoid similar problems, although the current method's responsibility is to find the executable file, but in the current implementation of the use of exec to execute command -v command to find, I think that to avoid the exec output error also belongs to the function needs to be implemented responsibilities
```bash 
sh: line 0: command: -r: invalid option
command: usage: command [-pVv] command [arg ...]
```

Reply:
I am sorry but I do not understand what you mean.

Reply:
I believe that since the current method uses the `exec()` function, then the `find()` method should have the responsibility to prevent exceptions from occurring with `exec()`.